src/plugins/irc-events/error.js

```diff
@@ -13,12 +13,51 @@ module.exports = function(irc, network) {
 			text = `${data.channel}: `;
 		}
 
-		if (data.error === "user_on_channel") {
-			text += `User (${data.nick}) is already on channel`;
-		} else if (data.reason) {
-			text += `${data.reason} (${data.error})`;
-		} else {
-			text += data.error;
+		switch (data.error) {
+			case "bad_channel_key":
+				text += "Bad channel key.";
+				break;
+			case "banned_from_channel":
+				text += "You have been banned from the channel.";
+				break;
+			case "cannot_send_to_channel":
+				text += "Cannot send to channel.";
+				break;
+			case "channel_is_full":
+				text += "The channel is full.";
+				break;
+			case "chanop_privs_needed":
+				text += "You're not a channel operator.";
+				break;
+			case "invite_only_channel":
+				text += "The channel is invite only.";
+				break;
+			case "no_such_nick":
+				text += `User (${data.nick}) hasn't logged in or does not exist.`;
+				break;
+			case "not_on_channel":
+				text += "You're not on the channel.";
+				break;
+			case "password_mismatch":
+				text += "Invalid password.";
+				break;
+			case "too_many_channels":
+				text += "You've already reached the maximum number of channels allowed.";
+				break;
+			case "user_not_in_channel":
+				text += `User (${data.nick}) is not on the channel.`;
+				break;
+			case "user_on_channel":
+				text += `User (${data.nick}) is already on the channel.`;
+				break;
+			default:
+				if (data.reason) {
+					text += `${data.reason} (${data.error})`;
+				} else {
+					text += data.error;
+				}
+
+				break;
 		}
 
 		const msg = new Msg({
```

Comment:
As I mentioned on IRC, the text formatting should be moved to the client.

Make an `error` messagetype component so the text can be formatted there. Remove `text` from the `Msg` and put `data.channel`, `data.reason`, `data.error` there instead, so the entire string formatting moves to the client.